cmd/deploy/deploy.go

```diff
@@ -595,11 +595,23 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 }
 
 func buildImages(ctx context.Context, build func(context.Context, *types.BuildOptions) error, getServicesToBuild func(context.Context, *model.Manifest, []string) ([]string, error), deployOptions *Options) error {
+	var stackServices map[string]bool
+
+	if stack := deployOptions.Manifest.GetStack(); stack != nil {
+		stackServices = stack.GetServicesWithBuildSection()
+	}
+
+	allServicesWithBuildSection := deployOptions.Manifest.GetBuildServices()
+	oktetoManifestBuildServices := setDifference(allServicesWithBuildSection, stackServices)
+
 	if deployOptions.Build {
+		finalStackIntersection := setIntersection(stackServices, sliceToSet(deployOptions.servicesToDeploy))
```

Comment:
I think this does the correct behavior, but it's complex to understand.

I'm thinking something like this might be simpler to understand:
```
...
allServicesWithBuildSection := deployOptions.Manifest.GetBuildServices()
oktetoManifestBuildServices := setDifference(allServicesWithBuildSection, stackServices)
servicesToDeployWithBuildSection := setIntersection(allServicesWithBuildSection, sliceToSet(deployOptions.servicesToDeploy))
servicesToBuildSet := setUnion(oktetoManifestBuildServices, servicesToDeployWithBuildSection)
```

Reply:
i am not sure i am understanding this

Reply:
If I understand this logic correctly, with Javi's change we'll build:
- `deployOptions.servicesToDeploy` that are defined in the stack and have a build section
- services with build section defined in the `okteto.yml` manifest

Since the `manifest`'s build section contains the `stack`'s build services, if there's a service from `deployOptions.servicesToDeploy` that's not in the stack, it will be in the manifest and will get built anyways. 
I think the logic is more clear if we compare everything to `allServicesWithBuildSection`. So in summary we'll build:
- `deployOptions.servicesToDeploy` that have a build section
- services with build section defined in the `okteto.yml` manifest